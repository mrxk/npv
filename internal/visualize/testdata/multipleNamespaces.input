---
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: one
  namespace: default
spec:
  egress:
  - ports:
    - port: 1111
      protocol: TCP
    to:
    - ipBlock:
        cidr: 0.0.0.0/0
  podSelector:
    matchLabels:
      app: app1
  policyTypes:
  - Egress
---
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: two
  namespace: one
spec:
  egress:
  - ports:
    - port: 2222
      protocol: TCP
    to:
    - ipBlock:
        cidr: 0.0.0.0/0
  podSelector:
    matchLabels:
      app: app1
  policyTypes:
  - Egress
---
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: three
  namespace: one
spec:
  egress:
  - ports:
    - port: 3333
      protocol: TCP
    to:
    - ipBlock:
        cidr: 0.0.0.0/0
  podSelector:
    matchLabels:
      app: app1
  policyTypes:
  - Egress
---
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: four
  namespace: two
spec:
  ingress:
  - ports:
    - port: 4444
      protocol: TCP
    from:
    - ipBlock:
        cidr: 0.0.0.0/0
  podSelector:
    matchLabels:
      app: app1
  policyTypes:
  - Ingress
---
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: five
  namespace: two
spec:
  ingress:
  - ports:
    - port: 5555
      protocol: TCP
    from:
    - ipBlock:
        cidr: 0.0.0.0/0
  podSelector:
    matchLabels:
      app: app1
  policyTypes:
  - Ingress
