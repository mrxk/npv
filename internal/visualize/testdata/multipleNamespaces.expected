@startuml
left to right direction
frame Pods {
component "Name: one\lNamespace: default\lMatch Labels:\l    app: app1\l" as defaultappapp1 {
    portout " " as defaultappapp1portout
}
component "Name: three, two\lNamespace: one\lMatch Labels:\l    app: app1\l" as oneappapp1 {
    portout " " as oneappapp1portout
}
component "Name: five, four\lNamespace: two\lMatch Labels:\l    app: app1\l" as twoappapp1 {
    port "4444 (TCP)" as 0.0.0.0_0TCP4444port
    port "5555 (TCP)" as 0.0.0.0_0TCP5555port
}
}
frame Ingress {
component "IPBlock:\l    0.0.0.0/0\l" as 0.0.0.0_0 {
    portout " " as 0.0.0.0_0ingressportout
}
}
0.0.0.0_0ingressportout --down[#green]--> 0.0.0.0_0TCP4444port
0.0.0.0_0ingressportout --down[#green]--> 0.0.0.0_0TCP5555port
frame Egress {
component "IPBlock:\l    0.0.0.0/0\l" as 0.0.0.0_0TCP1111 {
    port "1111 (TCP)" as 0.0.0.0_0TCP1111egressport
    port "2222 (TCP)" as 0.0.0.0_0TCP2222egressport
    port "3333 (TCP)" as 0.0.0.0_0TCP3333egressport
}
}
defaultappapp1portout --down[#green]--> 0.0.0.0_0TCP1111egressport
oneappapp1portout --down[#green]--> 0.0.0.0_0TCP2222egressport
oneappapp1portout --down[#green]--> 0.0.0.0_0TCP3333egressport
@enduml
